lib/rules/max-statements.js

```diff
@@ -84,9 +84,18 @@ module.exports = {
          */
         function reportIfTooManyStatements(node, count, max) {
             if (count > max) {
+                const messageEnd = " has too many statements ({{count}}). Maximum allowed is {{max}}.";
+                let name = "This function";
+
+                if (node.id) {
+                    name = `Function '${node.id.name}'`;
+                } else if (node.parent.type === "MethodDefinition" || node.parent.type === "Property") {
+                    name = `Function '${node.parent.key.name}'`;
+                }
```

Comment:
I'm going to just fix @mysticatea's comment for now, as I was just trying to clean up a previous PR for merging. If #7435 gets merged, we can refactor to use the new function.